### ISSUE TITLE
Suite crashes with JUnitException when a test engine in it is filtered out

### ISSUE THREAD

**Issue by D4nielLa** [junit-team/junit5#2838]
If a suite contains test classes running with different test engines and one of the test engines is filtered out by a `PostDiscoveryFilter`, a JUnitException is thrown and no test is executed.

## Steps to reproduce

Suppose we have the following four classes.
```java
public class ExcludedTestClass {

    @org.junit.jupiter.api.Test
    public void test() {
    }
}
```

```java
public class IncludedTestClass {

    @org.junit.Test
    public void aTest() {
    }
}
```

```java
import org.junit.platform.suite.api.SelectClasses;
import org.junit.platform.suite.api.Suite;

@Suite
@SelectClasses({ ExcludedTestClass.class, IncludedTestClass.class})
public class SuiteClass {

}
```

```java
import org.junit.platform.engine.FilterResult;
import org.junit.platform.engine.TestDescriptor;
import org.junit.platform.engine.support.descriptor.MethodSource;
import org.junit.platform.launcher.PostDiscoveryFilter;

public class Filter implements PostDiscoveryFilter {

    @Override
    public FilterResult apply(TestDescriptor object) {
        if (object.getSource().isPresent() && object.getSource().get() instanceof MethodSource)
            if (((MethodSource) object.getSource().get()).getJavaMethod().getName().startsWith("test")) {
                return FilterResult.excluded(null);
            }
        return FilterResult.included(null);
    }

}
```

The class `Filter` is registered as a `PostDiscoveryFilter`. When `SuiteClass` is executed, the test method `IncludedTestClass#aTest` should be executed. Instead, a `JUnitException` is thrown and no test is executed. 

The reason is that after filtering, the field `launcherDiscoveryResult` of the `SuiteTestDescriptor` still contains the `JupiterEngineDescriptor`, which should not be executed because all of its children were filtered out. Therefore, the `execute` method in the `EngineExecutionOrchestrator` tries to execute this empty `EngineDescriptor`, which leads to the `JUnitException`.

## Context

 - Used versions (Jupiter/Vintage/Platform): Jupiter 5.8.2, Vintage 5.8.2, Platform 1.8.2
 - Build Tool/IDE: Eclipse

**Comment by mpkorstanje** [comment]
Looks like the crash has unintentionally been fixed by #2822 as `SuiteTestDescriptor` execute now eventually creates a new test plan from the discovery request in the `EngineExecutionOrchestrator`.

https://github.com/junit-team/junit5/blob/f42118ebbde43816e1213b6f2c57a80585d69637/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineExecutionOrchestrator.java#L66-L74

The resulting test plan does however look a bit off, the excluded engine is now a root:

```
testPlan = {TestPlan@4159} 
 roots = {Collections$SynchronizedSet@4161}  size = 1
  0 = {TestIdentifier@4158} "TestIdentifier [uniqueId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-jupiter], parentId = null, displayName = 'JUnit Jupiter', legacyReportingName = 'JUnit Jupiter', source = null, tags = [], type = CONTAINER]"
 children = {ConcurrentHashMap@4162}  size = 3
  {UniqueId@4147} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]" -> {Collections$SynchronizedSet@4187}  size = 1
  {UniqueId@4188} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase]" -> {Collections$SynchronizedSet@4189}  size = 1
  {UniqueId@4190} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]" -> {Collections$SynchronizedSet@4191}  size = 1
 allIdentifiers = {ConcurrentHashMap@4160}  size = 4
  {UniqueId@4202} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-jupiter]" -> {TestIdentifier@4158} "TestIdentifier [uniqueId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-jupiter], parentId = null, displayName = 'JUnit Jupiter', legacyReportingName = 'JUnit Jupiter', source = null, tags = [], type = CONTAINER]"
  {UniqueId@4188} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase]" -> {TestIdentifier@4171} "TestIdentifier [uniqueId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase], parentId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage], displayName = 'JUnit4TestsTestCase', legacyReportingName = 'org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase', source = ClassSource [className = 'org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase', filePosition = null], tags = [], type = CONTAINER]"
  {UniqueId@4190} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]" -> {TestIdentifier@4166} "TestIdentifier [uniqueId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage], parentId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite], displayName = 'JUnit Vintage', legacyReportingName = 'JUnit Vintage', source = null, tags = [], type = CONTAINER]"
  {UniqueId@4203} "[engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase]/[test:aTest(org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase)]" -> {TestIdentifier@4177} "TestIdentifier [uniqueId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase]/[test:aTest(org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase)], parentId = [engine:junit-platform-suite]/[suite:org.junit.platform.suite.engine.testsuites.MultiEngineSuite]/[engine:junit-vintage]/[runner:org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase], displayName = 'aTest', legacyReportingName = 'aTest', source = MethodSource [className = 'org.junit.platform.suite.engine.testcases.JUnit4TestsTestCase', methodName = 'aTest', methodParameterTypes = ''], tags = [], type = TEST]"
```

**Comment by mpkorstanje** [comment]
Looks like `addInternal` assumes that test engines do not have parents:

https://github.com/junit-team/junit5/blob/main/junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java#L121-L133
